### PR TITLE
recreating domain models & adding DbSets

### LINKED ISSUE #4: Refactor Domain Layer (a.k.a. Core Layer)
For the current Domain Models, there are a lot of Attributes that are decorating properties for EF Core.

## Tasks to be done

- [x] Remove all attributes from the Domain Models
- [x] Add `DbSet`s to all Models
- [x] Create a EF Migration to acknowledge Db Schema

We will build on our new Folder Structure from now on. The old will be left for reference.

## References

- [EntityFramework Core Tutorial Site](https://www.entityframeworktutorial.net/efcore/conventions-in-ef-core.aspx)

## Dependent on

- #2 
- #3

### PR DESCRIPTION
**Pull request by saraabdelsalam**
i have created the first 2 tasks but when i try to add migration it fails saying couldn't load assembly and i couldn't figure out how to fix this error

### CODE REVIEW

File: src/Vezeeta.API/appsettings.Development.json [hunk 1]
```diff
@@ -6,6 +6,6 @@
     }
   },
   "ConnectionStrings": {
-    "DefaultConnection": "Data Source=(localdb)\\MSSQLLocalDB;Initial Catalog=Vezeeta_DB;Integrated Security=True"
+    "DefaultConnection": "Data Source=.;Initial Catalog=Vezeeta_DB;Integrated Security=True;Encrypt=True;TrustServerCertificate=True;"
```
Comment (YoussefWaelMohamedLotfy) [inline src/Vezeeta.API/appsettings.Development.json:9]: Which database are you using? The `LocalDb` that comes with VS, or you have an instance installed on your machine?

File: src/Vezeeta.Domain/Entities/Appointment.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Vezeeta.Domain.Enums;
+
+namespace Vezeeta.Domain.Entities
+{
+    public class Appointment
```
Comment (YoussefWaelMohamedLotfy) [inline src/Vezeeta.Domain/Entities/Appointment.cs]: For all entities, you should be inheriting from `BaseEntity<T>` where T is the datatype of the ID for your entity. This is to follow the DDD in a simple fashion. This edit is to be applied in all Entities except the `ApplicationUser`

File: src/Vezeeta.Domain/Entities/DiscountCode.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Vezeeta.Domain.Enums;
+
+namespace Vezeeta.Domain.Entities
+{
+    public class DiscountCode
```
Comment (YoussefWaelMohamedLotfy) [inline src/Vezeeta.Domain/Entities/DiscountCode.cs]: For all entities, you should be inheriting from `BaseEntity<T>` where T is the datatype of the ID for your entity.

File: src/Vezeeta.Domain/Entities/Doctor.cs [hunk 4]
```diff
@@ -1,8 +1,25 @@
...
 
-namespace Vezeeta.Domain.Entities;
-
-public sealed class Doctor : BaseEntity<string>
+namespace Vezeeta.Domain.Entities
 {
-    // TODO: Fill with properties
-}
\ No newline at end of file
+    public class Doctor
```
Comment (YoussefWaelMohamedLotfy) [inline src/Vezeeta.Domain/Entities/Doctor.cs]: For all entities, you should be inheriting from `BaseEntity<T>` where T is the datatype of the ID for your entity.

File: src/Vezeeta.Domain/Entities/Request.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Vezeeta.Domain.Enums;
+
+namespace Vezeeta.Domain.Entities
+{
+    public class Request
```
Comment (YoussefWaelMohamedLotfy) [inline src/Vezeeta.Domain/Entities/Request.cs]: For all entities, you should be inheriting from `BaseEntity<T>` where T is the datatype of the ID for your entity.

File: src/Vezeeta.Domain/Entities/Times.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Vezeeta.Domain.Entities
+{ 
+    public class Times
```
Comment (YoussefWaelMohamedLotfy) [inline src/Vezeeta.Domain/Entities/Times.cs]: For all entities, you should be inheriting from `BaseEntity<T>` where T is the datatype of the ID for your entity.

File: src/Vezeeta.Domain/Entities/specialization.cs [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Vezeeta.Domain.Entities
+{
+    public class Specialization
```
Comment (YoussefWaelMohamedLotfy) [inline src/Vezeeta.Domain/Entities/specialization.cs]: For all entities, you should be inheriting from `BaseEntity<T>` where T is the datatype of the ID for your entity.

File: src/Vezeeta.Infrastructure/Data/AppDbContext.cs [hunk 8]
```diff
@@ -1,16 +1,23 @@
...
     {
     }
 
-    public DbSet<Doctor> Doctors { get; set; }
+    DbSet<Doctor> Doctors { get; set; }
+    DbSet<Specialization> Specializations { get; set; }
+    DbSet<Times> Times { get; set; }
+    DbSet<Appointment> Appointments { get; set; }
+    DbSet<Request> Requests { get; set; }
+    DbSet<DiscountCode> DiscountCodes { get; set; }
```
Comment (YoussefWaelMohamedLotfy) [inline src/Vezeeta.Infrastructure/Data/AppDbContext.cs]: The `DbSet<T>` should be `public` in order to access them, currently you cannot as he default is `private`

#OOP_Encapsulation

File: src/Vezeeta.Infrastructure/Vezeeta.Infrastructure.csproj [hunk 9]
```diff
@@ -7,6 +7,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="8.0.0" />
```
Comment (YoussefWaelMohamedLotfy) [inline src/Vezeeta.Infrastructure/Vezeeta.Infrastructure.csproj]: You should remove this, as it is already referenced in `Domain Layer`. and this project references the `Domain Layer` so the `Microsoft.AspNetCore.Identity.EntityFrameworkCore` will be visible to it.

**Review by YoussefWaelMohamedLotfy — CHANGES_REQUESTED** [review]

File: src/Vezeeta.API/appsettings.Development.json (hunk 1 above)
Comment (saraabdelsalam) [reply src/Vezeeta.API/appsettings.Development.json:9]: I'm using sql server database that is installed on my laptop and i have access to deal with it via sql server management studio not the local db in vs

**Comment by saraabdelsalam** [comment]
i have created the migration for the schema successfully, added access modifiers for the dbsets & removed the Unused package in the infrastructure layer in the last commit

**Comment by YoussefWaelMohamedLotfy** [comment]
@saraabdelsalam Is the Schema Checked for Correctness? If yes, then we can proceed with the next task.

Also, review my commit for the changes regarding indentation and spacing. I would like that you follow the same approach for more readable files and unified coding standards (for the future and those will edit after you)

### Highlights:

- Consistent Spacing
- Respect Pascal Casing and Camel Casing

**Comment by saraabdelsalam** [comment]
@YoussefWaelMohamedLotfy  i have a question to make sure the schema is right 100%, i have made class doctor inheritance from application user and in the migration schema the doctors table is renamed to AspNetUsers with all of the properties related to application user class and doctor class , so i wonder is this correct or there's a problem and doctors should appear as a separated table

**Comment by YoussefWaelMohamedLotfy** [comment]
> @YoussefWaelMohamedLotfy i have a question to make sure the schema is right 100%, i have made class doctor inheritance from application user and in the migration schema the doctors table is renamed to AspNetUsers with all of the properties related to application user class and doctor class , so i wonder is this correct or there's a problem and doctors should appear as a separated table

Refer to [this doc](https://learn.microsoft.com/en-us/ef/core/modeling/inheritance) about EF Inheritance Strategies & [this one about Inheritance Stategies benchmarks](https://learn.microsoft.com/en-us/ef/core/performance/modeling-for-performance#inheritance-mapping) and let's discuss what did you understand and what would be your decision.

**Comment by YoussefWaelMohamedLotfy** [comment]
@saraabdelsalam Don't forget to resolve code comments above once it's done

**Pull request merged**